src/json-completion.ts

```diff
@@ -0,0 +1,815 @@
+  // prefix i.e. if the token begins with `"to`, then the
+  // label needs to have the quotes as well for it to match.
+  // However we are manually filtering the results so we can
+  // just use the unquoted key as the label, which is nicer
+  // and gives us more control.
+  // If no property value is present, then we add the colon as well.
+  // TODO: Use snippetCompletion to handle insert value + position cursor e.g. "key": "#{}"
+  // doc: https://codemirror.net/docs/ref/#autocomplete.snippetCompletion
+  // idea: https://discuss.codemirror.net/t/autocomplete-cursor-position-in-apply-function/4088/3
+  private getInsertTextForProperty(
+    key: string,
+    addValue: boolean,
+    propertySchema?: JSONSchema7Definition
+  ) {
+    let resultText = `"${key}"`;
+    if (!addValue) {
+      return resultText;
+    }
+    resultText += ': ';
+
+    let value;
+    let nValueProposals = 0;
+    if (typeof propertySchema === 'object') {
+      if (propertySchema.enum) {
+        if (!value && propertySchema.enum.length === 1) {
+          value = this.getInsertTextForGuessedValue(propertySchema.enum[0], '');
+        }
+        nValueProposals += propertySchema.enum.length;
+      }
+      if (typeof propertySchema.const !== 'undefined') {
+        if (!value) {
+          value = this.getInsertTextForGuessedValue(propertySchema.const, '');
+        }
+        nValueProposals++;
+      }
+      if (typeof propertySchema.default !== 'undefined') {
+        if (!value) {
+          value = this.getInsertTextForGuessedValue(propertySchema.default, '');
+        }
+        nValueProposals++;
+      }
+      if (
+        Array.isArray(propertySchema.examples) &&
+        propertySchema.examples.length
+      ) {
+        if (!value) {
+          value = this.getInsertTextForGuessedValue(
+            propertySchema.examples[0],
+            ''
+          );
+        }
+        nValueProposals += propertySchema.examples.length;
+      }
+      if (nValueProposals === 0) {
+        let type = Array.isArray(propertySchema.type)
+          ? propertySchema.type[0]
+          : propertySchema.type;
+        if (!type) {
+          if (propertySchema.properties) {
+            type = 'object';
+          } else if (propertySchema.items) {
+            type = 'array';
+          }
+        }
+        switch (type) {
+          case 'boolean':
+            value = '#{}';
+            break;
+          case 'string':
+            value = '"#{}"';
+            break;
+          case 'object':
+            value = '{#{}}';
+            break;
+          case 'array':
+            value = '[#{}]';
+            break;
+          case 'number':
+          case 'integer':
+            value = '#{0}';
+            break;
+          case 'null':
+            value = '#{null}';
+            break;
+          default:
+            return resultText;
+        }
+      }
+    }
+    if (!value || nValueProposals > 1) {
+      value = '$1';
```

Comment:
i think we just need to change this to `'#{}'` yes?

Reply:
Yup!